lib/src/ast.dart

```diff
@@ -233,15 +233,15 @@ File? locatePubspecFile(CompilationUnit compilationUnit) {
   }
 
   final file = resourceProvider.getFile(fullName);
-  var folder = file.parent;
+  var folder = file.parent2;
 
   // Look for a pubspec.yaml file.
-  while (folder != null) {
+  while (!folder.isRoot) {
```

Comment:
You could use `FolderExtension.withAncestors` here, e.g.
```dart
for (var current in file.parent2.withAncestors) {
  ...
}
```

Reply:
Nice!